components/match2/commons/starcraft_starcraft2/match_group_input_starcraft.lua

```diff
@@ -39,45 +38,68 @@ local OPPONENT_MODE_TO_PARTIAL_MATCH_MODE = {
 	team = 'team',
 	literal = 'literal',
 }
-local NOW = os.time(os.date('!*t') --[[@as osdateparam]])
-local TBD = 'tbd'
-local TBA = 'tba'
-
-local getStarcraftFfaInputModule = FnUtil.memoize(function()
-	return Lua.import('Module:MatchGroup/Input/Starcraft/Ffa')
-end)
 
----Module for converting input args of match group objects into LPDB records.
----This module is specific to the Starcraft and Starcraft2 wikis.
 local StarcraftMatchGroupInput = {}
+local MatchFunctions = {}
+local MapFunctions = {}
 
--- called from Module:MatchGroup
 ---@param match table
 ---@param options table?
 ---@return table
 function StarcraftMatchGroupInput.processMatch(match, options)
-	Table.mergeInto(
-		match,
-		StarcraftMatchGroupInput._readDate(match)
-	)
-	match = StarcraftMatchGroupInput._getTournamentVars(match)
 	if Logic.readBool(match.ffa) then
-		match = getStarcraftFfaInputModule().adjustData(match)
-	else
-		match = StarcraftMatchGroupInput._adjustData(match)
+		return DeprecatedCustomMatchGroupInput.processMatch(match, options)
 	end
-	match = StarcraftMatchGroupInput._checkFinished(match)
-	match = StarcraftMatchGroupInput._getVodStuff(match)
-	match = StarcraftMatchGroupInput._getLinks(match)
-	match = StarcraftMatchGroupInput._getExtraData(match)
+
+	Table.mergeInto(match, MatchFunctions.readDate(match.date))
+
+	local opponents = Array.mapIndexes(function(opponentIndex)
+		return MatchGroupInput.readOpponent(match, opponentIndex, OPPONENT_CONFIG)
+	end)
+	local games = MatchFunctions.extractMaps(match, opponents)
+
+	-- TODO: check how we can get rid of this legacy stuff ...
+	Array.forEach(opponents, function(opponent, opponentIndex)
+		if not Logic.readBool(opponent.win) then return end
+		opponent.win = nil
+		match.winner = match.winner or opponentIndex
+	end)
+
+	Array.forEach(opponents, MatchFunctions.addOpponentExtradata)
```

Comment:
check if needed
```suggestion
	Array.forEach(opponents, MatchFunctions.enrichFactionData)
	Array.forEach(opponents, MatchFunctions.addOpponentExtradata)
```
- faction handling for archon and other special types?
- if needed also move the isarchon out of the extradata function and into this function instead (still sets the value in extradata though)
- set default factions?
```lua
	Array.forEach(opponent.match2players, function(player)
		player.extradata = Table.merge({faction = Faction.defaultFaction}, player.extradata)
	end)
```